diff --git a/src/paymentdisclosure.h b/src/paymentdisclosure.h
index b4f56eb4..c0c2d3d2 100644
--- a/src/paymentdisclosure.h
+++ b/src/paymentdisclosure.h
@@ -71,7 +71,7 @@ struct PaymentDisclosurePayload {
     uint8_t version;        // 0 = experimental, 1 = first production version, etc.
     uint256 esk;            // zcash/NoteEncryption.cpp
     uint256 txid;           // primitives/transaction.h
-    size_t js;              // Index into CTransaction.vjoinsplit
+    uint64_t js;            // Index into CTransaction.vjoinsplit
     uint8_t n;              // Index into JSDescription fields of length ZC_NUM_JS_OUTPUTS
     libzcash::PaymentAddress zaddr; // zcash/Address.hpp
     std::string message;     // parameter to RPC call
diff --git a/src/serialize.h b/src/serialize.h
index f776d6d9..d6daf4dd 100644
--- a/src/serialize.h
+++ b/src/serialize.h
@@ -649,7 +649,11 @@ void Serialize_impl(Stream& os, const std::vector<T, A>& v, int nType, int nVers
 {
     WriteCompactSize(os, v.size());
     for (typename std::vector<T, A>::const_iterator vi = v.begin(); vi != v.end(); ++vi)
+#if defined(__APPLE__)
+        ::Serialize(os, static_cast<T>(*vi), nType, nVersion);
+#else
         ::Serialize(os, (*vi), nType, nVersion);
+#endif
 }
 
 template<typename Stream, typename T, typename A>
diff --git a/src/wallet/wallet.h b/src/wallet/wallet.h
index 2b481d87..0a497b3d 100644
--- a/src/wallet/wallet.h
+++ b/src/wallet/wallet.h
@@ -159,7 +159,7 @@ public:
     // Transaction hash
     uint256 hash;
     // Index into CTransaction.vjoinsplit
-    size_t js;
+    uint64_t js;
     // Index into JSDescription fields of length ZC_NUM_JS_OUTPUTS
     uint8_t n;
 
