diff --git a/Makefile b/Makefile
index 99a6930..bb3431f 100644
--- a/Makefile
+++ b/Makefile
@@ -40,10 +40,12 @@ CFLAGS += -D_BSD_SOURCE
 CFLAGS += -D_DEFAULT_SOURCE
 ifneq ($(OS), FreeBSD)
 ifneq ($(OS), Darwin)
+ifneq ($(OS), SunOS)
 #CFLAGS += -D_POSIX_C_SOURCE=200112L
 CFLAGS += -D_XOPEN_SOURCE=600
 endif
 endif
+endif
 #CFLAGS += -ansi
 # -DNO_GETADDRINFO
 INSTALL = install
diff --git a/minissdpc.c b/minissdpc.c
index b46add2..db315ef 100644
--- a/minissdpc.c
+++ b/minissdpc.c
@@ -70,6 +70,9 @@ struct sockaddr_un {
 
 #if !defined(HAS_IP_MREQN) && !defined(_WIN32)
 #include <sys/ioctl.h>
+#if defined(__sun)
+#include <sys/sockio.h>
+#endif
 #endif
 
 #if defined(HAS_IP_MREQN) && defined(NEED_STRUCT_IP_MREQN)
