diff --git a/Makefile b/Makefile
index 44875f0..b6ffba5 100644
--- a/Makefile
+++ b/Makefile
@@ -231,16 +231,8 @@ $(LIBGTEST_A): $(GTESTDIR)/src/gtest-all.cc $(DEPINST_EXISTS)
 	$(CXX) -o $(DEPINST)/lib/gtest-all.o   -I $(GTESTDIR) -c -isystem $(GTESTDIR)/include $< $(CXXFLAGS)
 	$(AR) -rv $(LIBGTEST_A) $(DEPINST)/lib/gtest-all.o
 
-# libsnark.a will contains all of our relevant object files, and we also mash in the .a files of relevant dependencies built by ./prepare-depends.sh
-$(LIBSNARK_A): $(LIB_OBJS) $(AR_LIBS)
-	( \
-	  echo "create $(LIBSNARK_A)"; \
-	  echo "addmod $(LIB_OBJS)"; \
-	  if [ -n "$(AR_LIBS)" ]; then for AR_LIB in $(AR_LIBS); do echo addlib $$AR_LIB; done; fi; \
-	  echo "save"; \
-	  echo "end"; \
-	) | $(AR) -M
-	$(AR) s $(LIBSNARK_A)
+$(LIBSNARK_A): $(LIB_OBJS)
+	$(AR) crs $@ $^
 
 libsnark.so: $(LIBSNARK_A) $(DEPINST_EXISTS)
 	$(CXX) -o $@   --shared -Wl,--whole-archive $(LIBSNARK_A) $(CXXFLAGS) $(LDFLAGS) -Wl,--no-whole-archive $(LDLIBS)
